app/query_objects/cafe_queries.rb

```diff
@@ -0,0 +1,62 @@
+class CafeQueries
+  # returns: all users who worked during study period(lp) and year
+  def self.working_users(lp, year)
+    join_cafe_shifts(User.joins(:cafe_shifts), lp, year).includes(:cafe_workers).uniq
+  end
+
+  # returns: all cafe_workers from study period and year
+  def self.cafe_workers(lp, year)
+    join_cafe_shifts(CafeWorker.joins(:cafe_shift), lp, year)
+  end
+
+  def self.highscore_groups(lp, year)
+    (join_cafe_shifts(CafeWorker.where.not(group: "").where.not(group: nil).
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.